### ISSUE TITLE
Automatically remove __future__ imports

### ISSUE THREAD

**Issue by charliermarsh** [charliermarsh/ruff#312]
This may require adding target Python versions to the configuration settings.

Source: https://github.com/asottile/pyupgrade

![Screen Shot 2022-10-03 at 6 19 54 PM](https://user-images.githubusercontent.com/1309177/193696375-8683f099-353d-4a92-9df4-d0e14aaa0847.png)

**Comment by jmg-duarte** [comment]
Hey Charlie! I'm interested in helping, can you provide me with some guidance on where to look to implement this?

**Comment by charliermarsh** [comment]
Yes absolutely! Will write something up now...

**Comment by charliermarsh** [comment]
**Step 1**: Create a new lint rule to flag these lines, something like `UnnecessaryFutureImports`. There are some instructions on adding a new `CheckCode` and `CheckKind` in the [`CONTRIBUTING.md`](https://github.com/charliermarsh/ruff/blob/main/CONTRIBUTING.md#example-adding-a-new-lint-rule).


**Step 2**: Add the actual logic to detect these errors. This will start in `src/check_ast.rs` -- look for the `if let Some("__future__") = module.as_deref() { ... }` block, where we handle future imports, and add something like:

```rust
if self.settings.enabled.contains(&CheckCode::YOUR_NEW_CHECK_CODE) {
    plugins::unnecessary_future_import(self, name);
}
```

Then, add a file in `src/plugins/unnecessary_future_imports.rs`. Take a look at some of the other examples therein. We'll need to base the exact detection on a check of `checker.settings.target_version`, since certain future imports are only unnecessary after certain Python versions.

(You might see some rules that go `ast/checks.rs` instead of the `plugins` system, but I prefer the `plugins` system for complex rules like these.)

**Step 3**: Add a test fixture to `resources/test/fixtures`, and a line to the `#[test_case(CheckCode::C405, Path::new("C405.py"); "C405")]` group in `linter.rs`. We typically create one fixture file per rule.

**Step 4**: Call `check.amend` to delete the statement. We actually might be able to use `src/autofix/fixes.rs#remove_unused_import_froms` for this? But this can be a separate PR, even Steps 1-3 would be awesome without the autofix.

**Comment by charliermarsh** [comment]
Any questions or confusion just ask here! I want to be helpful :)

**Comment by jmg-duarte** [comment]
Hey @charliermarsh! Sorry for the long time to reply, suddenly got very busy but should be slowing down now.

Your overview is amazing 🙌 thank you very much for the help!

For now, one question I got as soon as I started implementing was which code should I use. As `pyupgrade` only does in-place transformations.

**Comment by charliermarsh** [comment]
No worries! Thanks for looking into this and for your interest in the project :)

You can use U009 as the check code. I’m just auto-incrementing under the U prefix for  pyupgrade (since pyupgrade itself doesn’t have any concept of error codes).

**Comment by jmg-duarte** [comment]
Hey @charliermarsh (and others) I've been kept busy by other things and I haven't had time to handle this. 
Just wanted to let you and others know that if someone want to pick this up, it's okay since I can't guarantee I'll make a PR on time. 

Sorry about this!

**Comment by charliermarsh** [comment]
No worries! No obligation on your part of course. Appreciate the interest, hope you're able to contribute at some point in the future :)

**Comment by chammika-become** [comment]
@charliermarsh 👋  I attempted at this. Thanks for the awesome explanation above. I managed to get 1-3 steps with the above PR.
